### PR TITLE
Add support for exempting users from the merge window.

### PR DESCRIPTION
**Pull request by xrvdg**
This PR adds support for exemption from the merge window for whitelisted users. This is useful for bots that automatically merge low impact changes.

### CODE REVIEW

File: src/EventLoop.hs [hunk 1]
```diff
@@ -140,7 +141,7 @@ runLogicEventLoop
   -> (ProjectState -> m ())
   -> ProjectState
   -> m ProjectState
-runLogicEventLoop triggerConfig projectConfig runTime runGit runGithub getNextEvent publish initialState =
+runLogicEventLoop triggerConfig projectConfig mergeWindowExemption runTime runGit runGithub getNextEvent publish initialState =
```
Comment (FPtje) [inline src/EventLoop.hs:144]: Nit: This line and the line that sets `state1` are now a bit longer than 100 characters

File: src/Logic.hs [hunk 2]
```diff
@@ -296,15 +296,16 @@ clearPullRequest prId pr state =
...
   -> ProjectState
   -> Action ProjectState
-handleEventInternal triggerConfig projectConfig event = case event of
+handleEventInternal triggerConfig projectConfig mergeWindowExemption event = case event of
   PullRequestOpened pr branch baseBranch sha title author -> handlePullRequestOpened pr branch baseBranch sha title author
   PullRequestCommitChanged pr sha -> handlePullRequestCommitChanged pr sha
   PullRequestClosed pr            -> handlePullRequestClosedByUser pr
   PullRequestEdited pr title baseBranch -> handlePullRequestEdited pr title baseBranch
-  CommentAdded pr author body     -> handleCommentAdded triggerConfig projectConfig pr author body
+  CommentAdded pr author body     -> handleCommentAdded triggerConfig projectConfig mergeWindowExemption pr author body
```
Comment (FPtje) [inline src/Logic.hs:308]: Same nit about line length, although I see that the untouched case `PullRequestOpened` has it too. I think it's better to try to limit the line length to 100 characters if it's not too much hassle.

File: tests/Spec.hs [hunk 3]
```diff
@@ -804,6 +807,23 @@ main = hspec $ do
       fromJust (Project.lookupPullRequest prId state') `shouldSatisfy`
         (\pr -> Project.approval pr == Just (Approval (Username "deckard") Project.Merge 0))
 
+    it "test tag" $ do
```
Comment (FPtje) [inline tests/Spec.hs:810]: Thanks for adding a test! I assume this is testing the merge on friday from the exempted bot. Could you rename `test tag` to describe that specific test?

**Review by FPtje — APPROVED** [review]
Thanks for adding this option! There are some nits, but otherwise LGTM!

**Comment by xrvdg** [comment]
@OpsBotPrime merge

**Comment by OpsBotPrime** [comment]
Your merge request has been denied, because merging on Fridays is not recommended. To override this behaviour use the command `merge on Friday`.

**Comment by xrvdg** [comment]
@OpsBotPrime merge on Friday

**Comment by OpsBotPrime** [comment]
Pull request approved for merge by @xrvdg, rebasing now.

**Comment by OpsBotPrime** [comment]
Rebased as 3e908dcff8e80e0c8997469e9b20d5ec42bb0949, waiting for CI …

**Pull request merged**